lib/authorizer/oauth2.js

```diff
@@ -58,6 +59,34 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
+        const authId = auth.get('authId'),
+            refreshTokenHelper = auth.get('refreshTokenHelper');
+
+        // Go into the refresh token flow ONLY IF we have all necessary information present
+        if (authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken)) {
+            const refreshTokenPromise = refreshTokenHelper.refreshToken(authId);
```

Comment:
Why do we have a method `refreshToken` inside the `refreshTokenHelper`? Shouldn't `refreshTokenHelper` itself be a method?

Reply:
I've kept an interface as we may end up needing to expose more stuff here, especially with requests to support custom refresh flows with custom oauth spec coming up.